### PR TITLE
`Development`: Enhance architecture tests

### PR DESCRIPTION
**Pull request by MaximilianAnzinger**
#### General

- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Server
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

To maintain and improve the code quality of our project, solid architecture tests are necessary.

### Description

Refactors our existing architecture tests and adds multiple new ones.

#### Code Review
- [x] Code Review 1
- [ ] Code Review 2

### CODE REVIEW

File: src/test/java/de/tum/in/www1/artemis/AbstractArchitectureTest.java [hunk 1]
```diff
@@ -1,13 +1,31 @@
 package de.tum.in.www1.artemis;
```
Comment (Strohgelaender) [inline src/test/java/de/tum/in/www1/artemis/AbstractArchitectureTest.java:1]: Can you also move this into the `architecture` subpackage?

**Comment by MaximilianAnzinger** [comment]
TODO: enable `allPublicMethodsShouldReturnResponseEntity` check addressed in #8358

**Review by Strohgelaender — APPROVED** [review]
code looks good, thanks for splitting up the long `ArchtectureTest` class.

File: src/test/java/de/tum/in/www1/artemis/architecture/RepositoryArchitectureTest.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+/**
+ * This class contains architecture tests for the persistence layer.
+ */
+class RepositoryArchitectureTest extends AbstractArchitectureTest {
+
+    @Test
+    void shouldBeNamedRepository() {
+        ArchRule rule = classes().that().areAnnotatedWith(Repository.class).should().haveSimpleNameEndingWith("Repository")
+                .because("repositories should have a name ending with 'Repository'.");
+    }
```
Comment (Hialus) [inline src/test/java/de/tum/in/www1/artemis/architecture/RepositoryArchitectureTest.java:42]: You do not check this rule

**Review by Hialus — DISMISSED** [review]

**Review by Strohgelaender — APPROVED** [review]

**Review by bensofficial — APPROVED** [review]

**Review by Strohgelaender — APPROVED** [review]

**Review by Hialus — APPROVED** [review]

**Review by bensofficial — APPROVED** [review]

**Pull request merged**